### PR TITLE
adds test for issue#2767

### PR DESCRIPTION
**Pull request by hornpolish**
adds test for #2767

### CODE REVIEW

File: plugin/builtin/patchtransformer/PatchTransformer_test.go [hunk 1]
```diff
@@ -636,3 +636,68 @@ spec:
...
+      containers:
+      - image: test-image
+        name: test-deployment
+        ports:
+        - containerPort: 8080
+          name: take-over-the-world
+          protocol: TCP
+        - containerPort: 8080
+          name: disappearing-act
+          protocol: TCP
```
Comment (monopole) [inline plugin/builtin/patchtransformer/PatchTransformer_test.go:657]: nit: Please in-line this (to L682) and mention the issue number in the comment above the test.

Comment (hornpolish) [reply plugin/builtin/patchtransformer/PatchTransformer_test.go]: no probs; i think i did that.  power just came back - coastal NC had a good storm last night.

**Pull request merged**